snapshotter/snapshotter_id_ping.py

```diff
@@ -17,15 +17,58 @@ async def main():
     If snapshotting is allowed, sets the active status key in Redis to True and exits with code 0.
     If snapshotting is not allowed, sets the active status key in Redis to False and exits with code 1.
     """
-    aioredis_pool = RedisPoolCache(pool_size=1000)
-    await aioredis_pool.populate()
-    redis_conn = aioredis_pool._aioredis_pool
-    
-    await redis_conn.set(
-        active_status_key,
-        int(True),
-    )
-    sys.exit(0)
+    try:
+        aioredis_pool = RedisPoolCache(pool_size=1000)
+        await aioredis_pool.populate()
+        redis_conn = aioredis_pool._aioredis_pool
+
+        anchor_rpc = RpcHelper(settings.anchor_chain_rpc)
+        await anchor_rpc.init(redis_conn=redis_conn)
+        protocol_abi = read_json_file(settings.protocol_state.abi)
+        protocol_state_contract = anchor_rpc.get_current_node()['web3_client'].eth.contract(
+            address=Web3.toChecksumAddress(
+                settings.protocol_state.address,
+            ),
+            abi=protocol_abi,
+        )
+
+        snapshotter_address = to_checksum_address(settings.instance_id)
+
+        snapshotters_arr_query = await anchor_rpc.web3_call(
+            [
+                protocol_state_contract.functions.allSnapshotters(snapshotter_address),
```

Comment:
Do we need both checks? I think the slot_id check alone is sufficient. Let's not make extra RPC calls IMO.

Reply:
Both are not needed. I was using both calls to tailor the error message to the specific issue, but considering there are only two options, it doesn't seem worth it. Fixed it here:

https://github.com/PowerLoom/pooler/commit/62c1411deb360eba65c75cf034b2959bdfc77d24